spec/views/two_factor_authentication/otp_verification/show.html.slim_spec.rb

```diff
@@ -24,19 +24,34 @@
     it 'has a localized heading' do
       render
 
-      expect(rendered).to have_content @presenter.header
+      expect(rendered).to have_content t('devise.two_factor_authentication.header_text')
     end
 
-    it 'informs the user that an OTP has been sent to their number' do
+    it 'informs the user that an OTP has been sent to their number via #help_text' do
       build_stubbed(:user, :signed_up)
+
+      code_link = link_to(
+        t("links.two_factor_authentication.resend_code.#{@presenter.delivery_method}"),
+        @presenter.resend_code_path
+      )
+
+      help_text = t("instructions.2fa.#{@presenter.delivery_method}.confirm_code_html",
+                    number: @presenter.phone_number_tag,
+                    resend_code_link: code_link)
+
       render
 
-      expect(rendered).to include @presenter.help_text
+      expect(rendered).to include help_text
     end
 
-    it 'allows user to resend code' do
+    it 'allows user to resend code using the same delivery method' do
       render
-      expect(rendered).to have_link(nil, href: @presenter.resend_code_path)
+
+      resend_path = otp_send_path(otp_delivery_selection_form: {
+                                    otp_method: @presenter.delivery_method
+                                  }, resend: true)
+
+      expect(rendered).to have_link(nil, href: resend_path)
```

Comment:
Also, it looks like we're missing a scenario for resending the code when the delivery method is Voice. We should make sure there is a link with the text `t('links.two_factor_authentication.resend_code.voice')`.